gwswlib/hydx.py

```diff
@@ -0,0 +1,242 @@
+            "unit": "-",
+            "type": ('string', 3),
+            "required": True,
+        },
+        {
+            "csvheader": "KNP_VRM",
+            "fieldname": "HydxVormPut",
+            "unit": "-",
+            "type": ('string', 3),
+            "required": True,
+        },
+        {
+            "csvheader": "KNP_BOK",
+            "fieldname": "HydxNiveauBinnenonderkantPut",
+            "unit": "m",
+            "type": ('float', 8.2),
+            "required": True,
+        },
+        {
+            "csvheader": "KNP_BRE",
+            "fieldname": "HydxBreedte_diameterPutbodem",
+            "unit": "mm",
+            "type": ('float', 8),
+            "required": True,
+        },
+        {
+            "csvheader": "KNP_LEN",
+            "fieldname": "HydxLengtePutbodem",
+            "unit": "mm",
+            "type": ('float', 8),
+            "required": False,
+        },
+        {
+            "csvheader": "KNP_TYP",
+            "fieldname": "HydxTypeKnooppunt",
+            "unit": "-",
+            "type": ('string', 3),
+            "required": True,
+        },
+        {
+            "csvheader": "INI_NIV",
+            "fieldname": "HydxInitieleWaterstand",
+            "unit": "m",
+            "type": ('float', 8.2),
+            "required": False,
+        },
+        {
+            "csvheader": "STA_OBJ",
+            "fieldname": "HydxStatusObject",
+            "unit": "-",
+            "type": ('string', 3),
+            "required": False,
+        },
+        {
+            "csvheader": "AAN_MVD",
+            "fieldname": "HydxAannameMaaiveldhoogte",
+            "unit": "-",
+            "type": ('string', 3),
+            "required": False,
+        },
+        {
+            "csvheader": "ITO_IDE",
+            "fieldname": "HydxIdentificatieDefinitieIT_object",
+            "unit": "-",
+            "type": ('string', 10),
+            "required": False,
+        },
+        {
+            "csvheader": "ALG_TOE",
+            "fieldname": "HydxToelichtingRegel",
+            "unit": "-",
+            "type": ('string', 100),
+            "required": False,
+        },
+    ]
+
+    @classmethod
+    def csvheaders(cls):
+        return [field["csvheader"] for field in cls.FIELDS]
+
+    @classmethod
+    def fields(cls):
+        return cls.FIELDS
+
+    def __init__(self, codes):
+        self.node = {}
+        FIELDS = self.fields()
+        for field in self.fields():
+            
+            # set empty values to None
+            if codes[field["csvheader"]] == '':
+                codes[field["csvheader"]] = None
+
+            # set fields to defined data type and load into object
+            if field["type"][0] == "string" and codes[field["csvheader"]] is not None:
+                self.node[field["csvheader"]] = str(codes[field["csvheader"]])
+            elif field["type"][0] == "float" and codes[field["csvheader"]] is not None:
```

Comment:
Ik vind die `[0]` niet zo mooi staan. *Als* de lengte (de `8.2` of `3` die in `[1]` staat) echt nodig is moet 'ie misschien in een aparte key gestopt worden voor de leesbaarheid?